### PR TITLE
Make ChildNode and ParentNode extend Node again

### LINKED ISSUE #1087: ChildNode/ParentNode no longer extending Node
According to https://github.com/microsoft/TypeScript/issues/45266, there are some breaks around [`ChildNode` no longer being a subtype of `Node`](https://github.com/microsoft/TypeScript/commit/9ebe11c2d375bfb1c8fad10d705ffae09583c522#diff-dc0eab937d15e62545da3ed7b4f40ad6b24f15dd88fbc6ceda2bfb4ed8356eb0L3511). This was also mentioned in Google's 4.4 beta adoption thread (#1067).

<details>
<summary>
Some breaks here
</summary>

> # [ant-design/ant-design](https://github.com/ant-design/ant-design)
> ### [tsconfig.json](https://github.com/ant-design/ant-design/blob/015b70664317917969ef5580cbc2dfb533dc0c02/tsconfig.json)
>  - `error TS2359: The right-hand side of an 'instanceof' expression must be of type 'any' or of a type assignable to the 'Function' interface type.`
>    - [components/_util/scrollTo.ts#L26](https://github.com/ant-design/ant-design/blob/015b70664317917969ef5580cbc2dfb533dc0c02/components/_util/scrollTo.ts#L26)
>  - `error TS2339: Property 'nodeType' does not exist on type 'ChildNode'.`
>    - [components/typography/util.tsx#L143](https://github.com/ant-design/ant-design/blob/015b70664317917969ef5580cbc2dfb533dc0c02/components/typography/util.tsx#L143)
>    - [components/typography/util.tsx#L207](https://github.com/ant-design/ant-design/blob/015b70664317917969ef5580cbc2dfb533dc0c02/components/typography/util.tsx#L207)
>  - `error TS2345: Argument of type 'ChildNode' is not assignable to parameter of type 'Node'.`
>    - [components/typography/util.tsx#L160](https://github.com/ant-design/ant-design/blob/015b70664317917969ef5580cbc2dfb533dc0c02/components/typography/util.tsx#L160)
>    - [components/typography/util.tsx#L165](https://github.com/ant-design/ant-design/blob/015b70664317917969ef5580cbc2dfb533dc0c02/components/typography/util.tsx#L165)
>    - [components/typography/util.tsx#L220](https://github.com/ant-design/ant-design/blob/015b70664317917969ef5580cbc2dfb533dc0c02/components/typography/util.tsx#L220)
>  - `error TS2339: Property 'textContent' does not exist on type 'ChildNode'.`
>    - [components/typography/util.tsx#L227](https://github.com/ant-design/ant-design/blob/015b70664317917969ef5580cbc2dfb533dc0c02/components/typography/util.tsx#L227)

> # [sindresorhus/refined-github](https://github.com/sindresorhus/refined-github)
> ### [tsconfig.json](https://github.com/sindresorhus/refined-github/blob/4af1669e2dba94a1d170dfe563b88416582be716/tsconfig.json)
>  - `error TS2345: Argument of type 'Element | ChildNode' is not assignable to parameter of type 'string | Node'.`
>    - [source/helpers/dom-utils.ts#L48](https://github.com/sindresorhus/refined-github/blob/4af1669e2dba94a1d170dfe563b88416582be716/source/helpers/dom-utils.ts#L48)

</details>

In #1067, @saschanaz explained

> It's a mixin interface implemented by some Node interfaces. The correct way to receive Nodes that can be children is `ChildNode & Node`, just like `ParentNode & Node` for Nodes that can be parents.

I may be missing some context of why that's important to have - was there discussion around that? But regardless, I don't know if we can retroactively make that call. If that's a thing users need, does it make sense to create a `ChildNodeBase` and a `ParentNodeBase`?

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
Fixes #1087

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
CC @saschanaz

File: inputfiles/overridingTypes.jsonc [hunk 1]
```diff
@@ -1,6 +1,14 @@
 {
     "mixins": {
         "mixin": {
+            "ChildNode": {
+                "name": "ChildNode",
+                "extends": "Node"
```
Comment (saschanaz) [inline inputfiles/overridingTypes.jsonc:6]: TBH https://github.com/microsoft/TypeScript-DOM-lib-generator/issues/1087#issuecomment-891333791 is better to me with more spec compliance, why is this better?

File: inputfiles/overridingTypes.jsonc [hunk 2]
```diff
@@ -760,26 +768,8 @@
                 },
                 "properties": {
                     "property": {
-                        "parentNode": {
-                            "overrideType": "Node & ParentNode"
```
Comment (saschanaz) [inline inputfiles/overridingTypes.jsonc:764]: This should be ParentNode, same below

File: inputfiles/overridingTypes.jsonc (hunk 1 above)
Comment (saschanaz) [reply inputfiles/overridingTypes.jsonc:6]: BTW, in this way `name` is redundant.

Comment (DanielRosenwasser) [reply inputfiles/overridingTypes.jsonc:5]: I mentioned it in https://github.com/microsoft/TypeScript-DOM-lib-generator/issues/1087#issuecomment-891428070, but I don't think the `ChildNodeBase` and `ParentNodeBase` are strictly necessary and I'd rather be conservative about additions so close the RC.

Comment (saschanaz) [reply inputfiles/overridingTypes.jsonc:5]: >so close the RC.

Fair enough 😄

**Comment by saschanaz** [comment]
LGTM

**Pull request merged**